contrib/mesos/pkg/scheduler/executorinfo/registry.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executorinfo
+
+import (
+	"sync"
+
+	"fmt"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/mesos/mesos-go/mesosproto"
+	"k8s.io/kubernetes/contrib/mesos/pkg/node"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/meta"
+)
+
+// Registry is the interface that provides methods for interacting
+// with a registry of ExecutorInfo objects
+//
+// Get looks up an ExecutorInfo object for the given hostname
+//
+// New returns an ExecutorInfo object based on a given hostname and resources
+//
+// Invalidate invalidates the given hostname from this registry.
+// Note that a subsequent Get may recover the executor info.
+type Registry interface {
+	New(hostname string, resources []*mesosproto.Resource) *mesosproto.ExecutorInfo
+	Get(hostname string) (*mesosproto.ExecutorInfo, error)
+	Invalidate(hostname string)
+}
+
+// registry implements a map-based in-memory ExecutorInfo registry
+type registry struct {
+	cache *Cache
+	mu    sync.RWMutex // protects fields above
+
+	lookupNode node.LookupFunc
+	prototype  *mesosproto.ExecutorInfo
+}
+
+// NewRegistry returns a new executorinfo registry.
+// The given prototype is being used for properties other than resources.
+func NewRegistry(
+	lookupNode node.LookupFunc,
+	prototype *mesosproto.ExecutorInfo,
+	cache *Cache,
+) (Registry, error) {
```

Comment:
nit: i'm not sure how idiomatic it really is for constructors to return errors. though we probably do it in other places too I'm not sure that it's appropriate behavior. i'm under the impression that constructors are generally expected to panic if you pass in args that don't comply with the documented requirements.

thoughts?